platform-modules/batch-models/src/main/scala/org/ekstep/analytics/model/DeviceSummaryModel.scala

```diff
@@ -65,17 +67,33 @@ object DeviceSummaryModel extends IBatchModelTemplate[String, DeviceInput, Devic
             val dial_success = dialcodes_events.filter(f => f.edata.size > 0).length
             val dial_failure = dialcodes_events.filter(f => f.edata.size == 0).length
             val content_downloads = raw.filter(f => "INTERACT".equals(f.eid)).length
-            (index, DeviceSummary(index.device_id, index.channel, CommonUtil.roundDouble(total_ts, 2), total_launches, contents_played, unique_contents_played, content_downloads, DialStats(dial_count, dial_success, dial_failure), DtRange(startTimestamp, endTimestamp), syncts, startTimestamp))
+            (index, DeviceSummary(index.device_id, index.channel, CommonUtil.roundDouble(total_ts, 2), total_launches, contents_played, unique_contents_played, content_downloads, DialStats(dial_count, dial_success, dial_failure), DtRange(startTimestamp, endTimestamp), syncts, new Timestamp(startTimestamp)))
+        }
+        var firstAccessFromPostgres: RDD[FirstAccessByDeviceID] = null;
+        try {
+            firstAccessFromPostgres = summary.flatMap{x =>
+                val firstAccessQuery = s"SELECT device_id, first_access FROM $table WHERE device_id = '${x._1.device_id}'"
+                postgresDB.readFirstAccessFromDB(firstAccessQuery)
```

Comment:
Is this not executed on workers? Does it open a new connection for each request? Why can't we bulk get the data?